# Project 3

In [28]:
import plotly.io as pio

pio.renderers.default = "vscode+jupyterlab+notebook_connected"

In [29]:
import pandas as pd
import plotly.express as px

In [30]:
df = pd.read_csv('/Users/yutonomy/Dropbox/Columbia SIPA/24FALL/Computing in Context/project3/Neighborhood.csv')
df

Year Published  PUMA Borough                               Neighborhoods  \
0              2020  3701   Bronx          Riverdale, Fieldston & Kingsbridge   
1              2020  3701   Bronx          Riverdale, Fieldston & Kingsbridge   
2              2020  3701   Bronx          Riverdale, Fieldston & Kingsbridge   
3              2020  3701   Bronx          Riverdale, Fieldston & Kingsbridge   
4              2020  3701   Bronx          Riverdale, Fieldston & Kingsbridge   
..              ...   ...     ...                                         ...   
380            2020  4114  Queens  Far Rockaway, Breezy Point & Broad Channel   
381            2020  4114  Queens  Far Rockaway, Breezy Point & Broad Channel   
382            2020  4114  Queens  Far Rockaway, Breezy Point & Broad Channel   
383            2020  4114  Queens  Far Rockaway, Breezy Point & Broad Channel   
384            2020  4114  Queens  Far Rockaway, Breezy Point & Broad Channel   

                           CD  Join  NYC_Poverty_Rate  Median_Income  \
0     BX Community District 8  3701             0.152          30437   
1     BX Community District 8  3701             0.152          30437   
2     BX Community District 8  3701             0.152          30437   
3     BX Community District 8  3701             0.152          30437   
4     BX Community District 8  3701             0.152          30437   
..                        ...   ...               ...            ...   
380  QN Community District 14  4114             0.179          25920   
381  QN Community District 14  4114             0.179          25920   
382  QN Community District 14  4114             0.179          25920   
383  QN Community District 14  4114             0.179          25920   
384  QN Community District 14  4114             0.179          25920   

     Perc_White  Perc_Black  ...  Ind6Rank  \
0         0.324       0.124  ...      32.0   
1         0.324       0.124  ...      38.0   
2         0.324       0.124  ...      27.0   
3         0.324       0.124  ...      41.0   
4         0.324       0.124  ...      28.0   
..          ...         ...  ...       ...   
380       0.350       0.353  ...      42.0   
381       0.350       0.353  ...      55.0   
382       0.350       0.353  ...      15.0   
383       0.350       0.353  ...       NaN   
384       0.350       0.353  ...       NaN   

                                        Ind6Definition  \
0    Percentage of jobs in small businesses (less t...   
1    Percentage of bank & credit union branches off...   
2    Number of pre-foreclosure notices issued per 1...   
3        Average (mean) travel time to work in minutes   
4    Percentage of people eligible for SNAP receivi...   
..                                                 ...   
380  Number of pre-foreclosure notices issued per 1...   
381      Average (mean) travel time to work in minutes   
382  Percentage of people eligible for SNAP receivi...   
383                                                NaN   
384                                                NaN   

                           Ind7 Ind7Outcome Ind7Rank  \
0       Participatory Budgeting    0.071777     26.0   
1    Mobile Banking Utilization    0.205800     23.0   
2          Financial Counseling    0.000000     40.0   
3                           NaN         NaN      NaN   
4        Tax Credit Utilization    0.966000      9.0   
..                          ...         ...      ...   
380        Financial Counseling    1.413969     21.0   
381                         NaN         NaN      NaN   
382      Tax Credit Utilization    0.936000     25.0   
383                         NaN         NaN      NaN   
384                         NaN         NaN      NaN   

                                        Ind7Definition  \
0    Percentage of eligible residents casting a vot...   
1    Estimated percentage of residents that used mo...   
2    Full-Time Equivalent Financial Counselors per ...   
3           

- **Dataset(s) to be used:** [https://data.cityofnewyork.us/Business/Neighborhood-Financial-Health-Digital-Mapping-and-/r3dx-pew9/about_data]
- **Analysis question:** The relationship between race and average income: is there any relationship between the percentage of black people and the change in the average income?
- **Columns that will (likely) be used:**
  - Median_Income
  - Perc_Black
- **Hypothesis**: The bigger the percentage of the black people gets, the less the average income becomes
- **Site URL:** [https://github.com/yutonomy/project3]

# Check the data and clean the data if needed

In [31]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 385 entries, 0 to 384
Data columns (total 52 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Year Published    385 non-null    int64  
 1   PUMA              385 non-null    int64  
 2   Borough           385 non-null    object 
 3   Neighborhoods     385 non-null    object 
 4   CD                385 non-null    object 
 5   Join              385 non-null    int64  
 6   NYC_Poverty_Rate  385 non-null    float64
 7   Median_Income     385 non-null    int64  
 8   Perc_White        385 non-null    float64
 9   Perc_Black        385 non-null    float64
 10  Perc_Asian        385 non-null    float64
 11  Perc_Other        385 non-null    float64
 12  Perc_Hispanic     385 non-null    float64
 13  Goal              385 non-null    object 
 14  GoalName          385 non-null    object 
 15  GoalFullName      385 non-null    object 
 16  TotalOutcome      367 non-null    float64
 1

In [32]:
df.head()

Year Published  PUMA Borough                       Neighborhoods  \
0            2020  3701   Bronx  Riverdale, Fieldston & Kingsbridge   
1            2020  3701   Bronx  Riverdale, Fieldston & Kingsbridge   
2            2020  3701   Bronx  Riverdale, Fieldston & Kingsbridge   
3            2020  3701   Bronx  Riverdale, Fieldston & Kingsbridge   
4            2020  3701   Bronx  Riverdale, Fieldston & Kingsbridge   

                        CD  Join  NYC_Poverty_Rate  Median_Income  Perc_White  \
0  BX Community District 8  3701             0.152          30437       0.324   
1  BX Community District 8  3701             0.152          30437       0.324   
2  BX Community District 8  3701             0.152          30437       0.324   
3  BX Community District 8  3701             0.152          30437       0.324   
4  BX Community District 8  3701             0.152          30437       0.324   

   Perc_Black  ...  Ind6Rank  \
0       0.124  ...      32.0   
1       0.124  ...      38.0   
2       0.124  ...      27.0   
3       0.124  ...      41.0   
4       0.124  ...      28.0   

                                      Ind6Definition  \
0  Percentage of jobs in small businesses (less t...   
1  Percentage of bank & credit union branches off...   
2  Number of pre-foreclosure notices issued per 1...   
3      Average (mean) travel time to work in minutes   
4  Percentage of people eligible for SNAP receivi...   

                         Ind7 Ind7Outcome Ind7Rank  \
0     Participatory Budgeting    0.071777     26.0   
1  Mobile Banking Utilization    0.205800     23.0   
2        Financial Counseling    0.000000     40.0   
3                         NaN         NaN      NaN   
4      Tax Credit Utilization    0.966000      9.0   

                                      Ind7Definition  \
0  Percentage of eligible residents casting a vot...   
1  Estimated percentage of residents that used mo...   
2  Full-Time Equivalent Financial Counselors per ...   
3                                                NaN   
4     Percentage of eligible Tax Units claiming EITC   

                              Ind8 Ind8Outcome  Ind8Rank  \
0                              NaN         NaN       NaN   
1                              NaN         NaN       NaN   
2                              NaN         NaN       NaN   
3                              NaN         NaN       NaN   
4  Adults With High School Diploma    0.845032      21.0   

                                      Ind8Definition  
0                                                NaN  
1                                                NaN  
2                                                NaN  
3                                                NaN  
4  Percentage of adults with a high school degree...  

[5 rows x 52 columns]

In [33]:
df.sample(10)

Year Published  PUMA    Borough  \
307            2020  4103     Queens   
69             2020  3710      Bronx   
377            2020  4113     Queens   
24             2020  3704      Bronx   
135            2020  3809  Manhattan   
110            2020  3805  Manhattan   
236            2020  4010   Brooklyn   
354            2020  4110     Queens   
89             2020  3803  Manhattan   
26             2020  3802  Manhattan   

                                      Neighborhoods  \
307              Flushing, Murray Hill & Whitestone   
69                  Hunts Point, Longwood & Melrose   
377                       Howard Beach & Ozone Park   
24            Pelham Parkway, Morris Park & Laconia   
135                     Chinatown & Lower East Side   
110                                 Upper East Side   
236                 East Flatbush, Farragut & Rugby   
354            Ridgewood, Glendale & Middle Village   
89                                   Central Harlem   
26   Hamilton Heights, Manhattanville & West Harlem   

                               CD  Join  NYC_Poverty_Rate  Median_Income  \
307       QN Community District 7  4103             0.252          25517   
69   BX Community Districts 1 & 2  3710             0.293          15510   
377      QN Community District 10  4113             0.189          27016   
24       BX Community District 11  3704             0.205          25358   
135       MN Community District 3  3809             0.180          25667   
110       MN Community District 8  3805             0.073          71933   
236      BK Community District 17  4010             0.190          27248   
354       QN Community District 5  4110             0.186          31420   
89       MN Community District 10  3803             0.211          26075   
26        MN Community District 9  3802             0.235          24561   

     Perc_White  Perc_Black  ...  Ind6Rank  \
307       0.256       0.021  ...      13.0   
69        0.016       0.282  ...      37.0   
377       0.216       0.151  ...      45.0   
24        0.218       0.215  ...      33.0   
135       0.331       0.079  ...       6.0   
110       0.761       0.028  ...       NaN   
236       0.022       0.873  ...       3.0   
354       0.531       0.015  ...      22.0   
89        0.131       0.571  ...      50.0   
26        0.286       0.215  ...       NaN   

                                        Ind6Definition  \
307  Number of pre-foreclosure notices issued per 1...   
69   Number of pre-foreclosure notices issued per 1...   
377  Percentage of people eligible for SNAP receivi...   
24   Percentage of bank & credit union branches off...   
135      Average (mean) travel time to work in minutes   
110                                                NaN   
236  Percentage of people eligible for SNAP receivi...   
354  Number of pre-foreclosure notices issued per 1...   
89   Percentage of bank & credit union branches off...   
26                                                 NaN   

                           Ind7 Ind7Outcome Ind7Rank  \
307        Financial Counseling    0.832343     24.0   
69         Financial Counseling    7.737565      3.0   
377      Tax Credit Utilization    0.911000     39.0   
24   Mobile Banking Utilization    0.178800     34.0   
135                         NaN         NaN      NaN   
110                         NaN         NaN      NaN   
236      Tax Credit Utilization    0.968000      8.0   
354        Financial Counseling    0.000000     40.0   
89   Mobile Banking Utilization    0.198300     28.0   
26                          NaN         NaN      NaN   

                                        Ind7Definition  \
307  Full-Time Equivalent Financial Counselors per ...   
69   Full-Time Equivalent Financial Counselors per ...   
377     Percentage of eligible Tax Units claiming EITC   
24   Estimated percentage of residents that used mo...   
135                                                NaN   
110       

# Data Visualization
We will do the data visualization: visualize in the bar graph and the scatterplot then figure out the relationship or correlation between % of black people and Median Incomes.

In [38]:
fig = px.bar(df, x='Perc_Black', y='Median_Income', color_discrete_sequence=['#000000'])  # Pure black color 
fig.show()


In [35]:
fig = px.scatter(df, x='Perc_Black', y='Median_Income')
fig.show()

# Reflection
Based on the bar graph and the scatterplot above, it can be said that the higher the percentage of Black people, the lower the median income tends to be, and conversely, the lower the percentage, the higher the median income.

# Check the regression

In [36]:
import numpy as np
x = df['Perc_Black']
y = df['Median_Income']

coefficients = np.polyfit(x, y, 1)

print(f"Regression coefficient (slope): {coefficients[0]}")
print(f"Intercept: {coefficients[1]}")

Regression coefficient (slope): -20276.573844372164
Intercept: 35800.935109920705


# Takeaway
The regression coefficient (slope): -20276.57 indicates that for every 1% increase in Perc_Black (percentage of the Black population), the Median_Income (median income) decreases by approximately $20,276. Since the value is negative, it suggests a negative linear relationship between Perc_Black and Median_Income.